satellite/console/consoleweb/utils.go

```diff
@@ -49,20 +52,22 @@ func getToken(req *http.Request) string {
 }
 
 // getQuery retrieves graphql query from request
-func getQuery(req *http.Request) (query graphqlJSON, err error) {
+func getQuery(writer http.ResponseWriter, req *http.Request) (query graphqlJSON, err error) {
 	switch req.Method {
 	case http.MethodGet:
 		query.Query = req.URL.Query().Get(consoleql.Query)
 		return query, nil
 	case http.MethodPost:
-		return queryPOST(req)
+		return queryPOST(writer, req)
 	default:
 		return query, errs.New("wrong http request type")
 	}
 }
 
 // queryPOST retrieves graphql query from POST request
-func queryPOST(req *http.Request) (query graphqlJSON, err error) {
+func queryPOST(writer http.ResponseWriter, req *http.Request) (query graphqlJSON, err error) {
+	req.Body = http.MaxBytesReader(writer, req.Body, ContentLengthLimit)
```

Comment:
I guess `http.Request` may be required to be immutable. 
Anyway, if you are modifying `*http.Request` you can skip the propagation and do it in the handler itself. 
We can also modify code to use only `io.Reader` and `URL` object

Reply:
https://golang.org/pkg/net/http/#Handler
``Except for reading the body, handlers should not modify the provided Request.``
This is the case when I'm reading the body.
Everything else left unchanged.

Reply:
`req.Body = ...` changes the request.

Reply:
ok

Reply:
This is refers to reading the body.
According to `http` documentation, prohibited any other request changing except for body reading.

Reply:
```
xyz := http.MaxBytesReader(writer, req.Body, ContentLengthLimit), <-- this part is fine, this doesn't change the request
req.Body = xyz <-- this is not, because it's changing the request